WordPress/src/main/java/org/wordpress/android/ui/posts/PostsActivity.java

```diff
@@ -503,14 +414,99 @@ public void onSaveInstanceState(Bundle outState) {
         super.onSaveInstanceState(outState);
     }
 
-    public void setRefreshing(boolean refreshing) {
-        mPostList.setRefreshing(refreshing);
+    public class deletePostTask extends AsyncTask<Post, Void, Boolean> {
+        Post post;
+
+        @Override
+        protected void onPreExecute() {
+            // pop out of the detail view if on a smaller screen
+            popPostDetail();
+            showDialog(ID_DIALOG_DELETING);
+        }
+
+        @Override
+        protected void onPostExecute(Boolean result) {
+            if (result) {
+                WordPress.wpDB.deletePost(post);
+            }
+            if (mLoadingDialog == null || isFinishing()) {
+                return;
+            }
+            dismissDialog(ID_DIALOG_DELETING);
+            attemptToSelectPost();
+            if (result) {
+                Toast.makeText(PostsActivity.this, getResources().getText((mIsPage) ?
+                                R.string.page_deleted : R.string.post_deleted),
+                        Toast.LENGTH_SHORT).show();
+                requestPosts();
+                mPostList.requestPosts(false);
+                mPostList.setRefreshing(true);
+            } else {
+                AlertDialog.Builder dialogBuilder = new AlertDialog.Builder(PostsActivity.this);
+                dialogBuilder.setTitle(getResources().getText(R.string.connection_error));
+                dialogBuilder.setMessage(mErrorMsg);
+                dialogBuilder.setPositiveButton("OK",
+                        new DialogInterface.OnClickListener() {
+                            public void onClick(DialogInterface dialog, int whichButton) {
+                                // Just close the window.
```

Comment:
That's existing code, not something I touched in this PR, so I'll just
leave it as is.
On Jun 9, 2015 5:31 PM, "Tony Rankin" <notifications@github.com> wrote:

> In
> WordPress/src/main/java/org/wordpress/android/ui/posts/PostsActivity.java
> <https://github.com/wordpress-mobile/WordPress-Android/pull/2802#discussion_r32066700>
> :
>
> > +            attemptToSelectPost();
> > +            if (result) {
> > +                Toast.makeText(PostsActivity.this, getResources().getText((mIsPage) ?
> > +                                R.string.page_deleted : R.string.post_deleted),
> > +                        Toast.LENGTH_SHORT).show();
> > +                requestPosts();
> > +                mPostList.requestPosts(false);
> > +                mPostList.setRefreshing(true);
> > +            } else {
> > +                AlertDialog.Builder dialogBuilder = new AlertDialog.Builder(PostsActivity.this);
> > +                dialogBuilder.setTitle(getResources().getText(R.string.connection_error));
> > +                dialogBuilder.setMessage(mErrorMsg);
> > +                dialogBuilder.setPositiveButton("OK",
> > +                        new DialogInterface.OnClickListener() {
> > +                            public void onClick(DialogInterface dialog, int whichButton) {
> > +                                // Just close the window.
>
> Passing null will have the same effect here, your call if it's worth a
> commit or not.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/wordpress-mobile/WordPress-Android/pull/2802/files#r32066700>
> .
>